src/main/java/io/vertx/core/http/impl/Http1xClientConnection.java

```diff
@@ -181,7 +182,13 @@ private HttpRequest createRequest(
     }
     if (options.isTryUseCompression() && request.headers().get(ACCEPT_ENCODING) == null) {
       // if compression should be used but nothing is specified by the user support deflate and gzip.
-      request.headers().set(ACCEPT_ENCODING, DEFLATE_GZIP);
+      CharSequence acceptEncoding = DEFLATE_GZIP;
+      if (Brotli.isAvailable()) {
```

Comment:
Could you isolate this call in its own method? The idea is that it would ease writing a substitution is needed (the substitute would return `false`)

Reply:
yes that's fine to do this